### PR TITLE
[Ogre] update to 1.11.3

### PR DESCRIPTION
**Pull request by twMr7**
merge and re-PR

1. update ogre to upstream version 1.11.3
2. workaround patch to replace optimized/debug keyword by generator expression, which prevent cmake from generating wrong linking target to "optimized.lib"
3. @ras0219-msft 's modernization

### CODE REVIEW

File: ports/ogre/portfile.cmake [hunk 1]
```diff
@@ -95,33 +89,31 @@ if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
...
+        else()
+            file(RENAME ${CURRENT_PACKAGES_DIR}/lib/OgreMainStatic.lib ${CURRENT_PACKAGES_DIR}/lib/manual-link/OgreMainStatic.lib)
+        endif()
+    endif()
+    if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
+        file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/debug/lib/manual-link)
+        if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
+            file(RENAME ${CURRENT_PACKAGES_DIR}/debug/lib/OgreMain_d.lib ${CURRENT_PACKAGES_DIR}/debug/lib/manual-link/OgreMain_d.lib)
+        else()
+            file(RENAME ${CURRENT_PACKAGES_DIR}/lib/OgreMainStatic_d.lib ${CURRENT_PACKAGES_DIR}/lib/manual-link/OgreMainStatic_d.lib)
```
Comment (Rastaban) [inline ports/ogre/portfile.cmake]: Should this be /debug/lib/ instead of just /lib/?

Comment (Rastaban) [reply ports/ogre/portfile.cmake]: adding debug fixed a x64-windows-static build failure so I am adding that change

**Comment by Rastaban** [comment]
rerunning regression tests, if they all pass I will merge this PR

File: ports/ogre/portfile.cmake (hunk 1 above)
Comment (twMr7) [reply ports/ogre/portfile.cmake]: @Rastaban , you are right about that, thank you.

**Pull request merged**